### PR TITLE
Attempt to infer if a CMap file actually contains just a standard Identity-H/Identity-V map

### PR DESCRIPTION
**Pull request by Snuffleupagus**
Fixes _most_ of issue #5801, ~~and depends on PR #5713~~.

The reason for the gibberish characters is that the PDF file includes `toUnicode` entries with CMap files which manually build `Identity-H` maps, instead of simply referring to them by name. Hence this patch attempts to detect when that happens, to correctly classify such CMaps as `Identity-H`/`Identity-V`, thus ensuring that the `isIdentityUnicode` check actually works at [fonts.js#L2495](https://github.com/mozilla/pdf.js/blob/master/src/core/fonts.js#L2495).

Even with these patches, a couple of the characters are unfortunately still not rendered correctly (at least on Windows). The troublesome font is a non-embedded `CIDFontType2` version of `ArialBlack`, and it turns out that the glyph map for this font, for some inexplicable reason, differs _ever so slightly_ from the glyph map we include for "standard" fonts (see [`GlyphMapForStandardFonts`](https://github.com/mozilla/pdf.js/blob/master/src/core/fonts.js#L439)). This is really strange, since all the other versions of e.g. `Arial` and `Times` all seem to agree on _one_ glyph map.

As far as I can tell, issues with non-embedded `ArialBlack` fonts seem to be rare (and it might be platform specific too), which is why I'm not attempting to fix the display of the remaining incorrect characters. To do so we could probably include a supplemental glyph map, `GlyphMapForAlmostStandardFonts` perhaps :-), specially for `ArialBlack`. However, in the event that we want to do this, I think that can be done in a follow-up PR instead.

**Edit:** Unit tests added/improved.

### CODE REVIEW

File: src/core/cmap.js [hunk 1]
```diff
@@ -303,13 +304,28 @@ var CMap = (function CMapClosure() {
...
+    },
+
+    get isIdentityCMap() {
+      if (!(this.name === 'Identity-H' || this.name === 'Identity-V')) {
+        return false;
+      }
+      if (this._map.length !== 0x10000) {
+        return false;
+      }
+      for (var i = 0; i <= 0xFFFF; i++) {
```
Comment (brendandahl) [inline src/core/cmap.js]: Tiny nit: it's a bit strange to use 0x1000 above but then use 0xFFFF here. Can we just make it < 0xFFFF?

**Comment by brendandahl** [comment]
Good to makeref and merge once nit is addressed.

**Comment by Snuffleupagus** [comment]
Nit addressed.

/botio test

**Pull request merged**